tests/ga/test_collect_logs.py

```diff
@@ -0,0 +1,192 @@
+        # reuse a previous state
+        clear_singleton_instances(ProtocolUtil)
+
+    def tearDown(self):
+        if os.path.exists(self.archive_path):
+            os.remove(self.archive_path)
+        self.mock_archive_path.stop()
+        AgentTestCase.tearDown(self)
+
+    def _create_dummy_archive(self, size=1024):
+        with open(self.archive_path, "wb") as f: # pylint: disable=C0103
+            f.truncate(size)
+
+    def test_it_should_invoke_all_periodic_operations(self):
+        invoked_operations = []
+
+        with _create_collect_logs_handler() as collect_logs_handler:
+            def mock_run(self):
+                invoked_operations.append(self._name)
+
+            with patch.object(PeriodicOperation, "run", side_effect=mock_run, spec=CollectLogsHandler.run):
+                collect_logs_handler.run_and_wait()
+
+                expected_operations = ["collect_and_send_logs"]
+
+                self.assertEqual(invoked_operations.sort(), expected_operations.sort(),
+                                 "The collect logs thread did not invoke the expected operations")
+
+    @patch("azurelinuxagent.ga.collect_logs.CollectLogsHandler.collect_and_send_logs")
+    def test_it_should_only_collect_logs_if_conditions_are_met(self, patch_collect_and_send_logs):
+        # In order to collect logs, three conditions have to be met:
+        # 1) the flag must be set to true in the conf file
+        # 2) systemd must be managing services
+        # 3) python version 2.7+ which is automatically true for these tests since they are disabled on py2.6
+
+        with _create_collect_logs_handler(systemd_present=False) as collect_logs_handler:
+            with patch("azurelinuxagent.ga.collect_logs.conf.get_collect_logs", return_value=False):
+                collect_logs_handler.run_and_wait()
+
+        self.assertEqual(patch_collect_and_send_logs.call_count, 0, "Log collection should not have been enabled")
+
+        with _create_collect_logs_handler(systemd_present=False) as collect_logs_handler:
+            with patch("azurelinuxagent.ga.collect_logs.conf.get_collect_logs", return_value=False):
+                collect_logs_handler.run_and_wait()
+
+        self.assertEqual(patch_collect_and_send_logs.call_count, 0, "Log collection should not have been enabled")
+
+        with _create_collect_logs_handler(systemd_present=False) as collect_logs_handler:
+            with patch("azurelinuxagent.ga.collect_logs.conf.get_collect_logs", return_value=True):
+                collect_logs_handler.run_and_wait()
+
+        self.assertEqual(patch_collect_and_send_logs.call_count, 0, "Log collection should not have been enabled")
+
+        with _create_collect_logs_handler() as collect_logs_handler:
+            collect_logs_handler.run_and_wait()
+
+        self.assertEqual(patch_collect_and_send_logs.call_count, 1, "Log collection should have been enabled")
+
+    def test_log_collection_is_invoked_with_resource_limits(self):
```

Comment:
Probably is, but that wouldn't be a unit test... We should stick to testing our code, not systemd's inner workings I think.

Currently I have tested this (verified resource limiting works) manually. In the future, it might be worth adding a DCR test for this.